R/html_document.R

```diff
@@ -689,4 +719,49 @@ navbar_link_text <- function(x, ...) {
     tagList(x$text, ...)
 }
 
+add_anchor_sections <- function(anchor_sections, section_divs = FALSE) {
+
+  res <- list(args = NULL, lua_filters = NULL, extra_dependencies = NULL)
 
+  if (xfun::isFALSE(anchor_sections)) return(res)
+
+  if (!pandoc2.0()) {
+    stop("Using anchor_sections requires Pandoc 2.0+", call. = FALSE)
+  }
+
+  # default style used
+  style <- "hash"
+
+    # allowed customization
+  allowed_args <- c("style", "depth")
+  if (is.list(anchor_sections)) {
+    # check list elements
+    all_allowed <- all(names(anchor_sections) %in% allowed_args)
+    if (xfun::isFALSE(all_allowed)) {
+      stop("`anchor_sections` should be TRUE or one of [",
```

Comment:
```suggestion
      stop("`anchor_sections` should be TRUE or a list with names [",
```
?

Reply:
It can also be `FALSE` — overall I find the logic of this function a little hard to parse. I think it would be clearer if organised around the three permitted inputs:

* `FALSE`
* `TRUE`
* A list with names in `allowed_args`

This might require extracting out some helper functions to reduce duplication

Reply:
At first `anchor_sections = FALSE` was meaning "do not run the code of adding anchor section because all was in `html_document` as we do in other places in the body. 

Later on I find it better to refactor in a special helper `add_anchor_sections` but still need to check FALSE before for adding the components (this code: https://github.com/rstudio/rmarkdown/pull/1964/files#diff-93d1e6747a8a5a21e4668d9a67b3889d481426747e8a0f18d7b2ebfb32ee8413R382) 

So I need to choose where to check FALSE (related to your review above) 

1. in `html_document` body
2. in the helper itself 

Maybe 2 is clearer and I should restructure as you are suggesting